src/Compilers/CSharp/Portable/Syntax/Syntax.xml

```diff
@@ -1798,6 +1798,40 @@
       <summary>Creates an ImplicitStackAllocArrayCreationExpressionSyntax node.</summary>
     </FactoryComment>
   </Node>
+  <Node Name="CollectionExpressionSyntax" Base="ExpressionSyntax">
+    <Kind Name="CollectionExpression"/>
+    <Field Name="OpenBracketToken" Type="SyntaxToken">
+      <Kind Name="OpenBracketToken"/>
+    </Field>
+    <Field Name="Elements" Type="SeparatedSyntaxList&lt;CollectionElementSyntax&gt;" AllowTrailingSeparator="true">
+      <PropertyComment>
+        <summary>SeparatedSyntaxList of CollectionElementSyntax representing the list of elements in the collection expression.</summary>
+      </PropertyComment>
+    </Field>
+    <Field Name="CloseBracketToken" Type="SyntaxToken">
+      <Kind Name="CloseBracketToken"/>
+    </Field>
+  </Node>
+  <AbstractNode Name="CollectionElementSyntax" Base="CSharpSyntaxNode" />
+  <Node Name="ExpressionElementSyntax" Base="CollectionElementSyntax">
```

Comment:
i considered that all these subtypes keep Collection in the name, but it was very verbose.  e.x. ExpressionCollectionElementSyntax or CollectionExpressionElementSyntax.  And it got worse with things like DictionaryCollectionElementSyntax or COllectionDictionaryElementSyntax where you have both Collection and Dictionary.

The idea of XXXElementSyntax, where XXX was just Expression/Spread/Dictionary was nice.  Though a potential downside here is if we ever have some other feature that wants to use XXXELementSyntax and potentially may also want an ExpressionELement form of that.